### ISSUE TITLE
colorbar=True is missing text values when logz=True

### ISSUE THREAD

**Issue by jbednar** [ioam/holoviews#2606]
I can enable a colorbar for a logz plot, but it doesn't show any numbers, making it not be very useful:

```
import numpy as np, pandas as pd, holoviews as hv
hv.extension('bokeh')

url     = 'https://raw.githubusercontent.com/blmoore/blogR/master/data/measles_incidence.csv'
data    = pd.read_csv(url, skiprows=2, na_values='-')
yearly  = data.drop('WEEK', axis=1).groupby('YEAR').sum().reset_index()
measles = pd.melt(yearly, id_vars=['YEAR'], var_name='State', value_name='Incidence')

heatmap = hv.HeatMap(measles, label='Measles Incidence').options( \
    height=500, tools=['hover'], logz=True, invert_yaxis=True, labelled=[], 
    toolbar='above', xaxis=None, colorbar=True)

heatmap.opts(plot=dict(width=900))
```

![image](https://user-images.githubusercontent.com/1695496/39134330-04556354-46dc-11e8-85e9-9cf63a91f278.png)

Works fine without logz:

![image](https://user-images.githubusercontent.com/1695496/39134387-1f376262-46dc-11e8-9be6-1139fd91b498.png)

**Comment by philippjfr** [comment]
This is specifically because the lower bound of the range is zero which is not supported by the log colormapper.

**Comment by jlstevens** [comment]
Maybe this could show a warning in this case? Taking the log of zero is a reasonable thing to complain about...

**Comment by jlstevens** [comment]
I assume there is code to drop the labels for all invalid values, e.g negative logs as well. Maybe a warning show up for any such invalid condition..

**Comment by jbednar** [comment]
Presumably there is code elsewhere to deal with the zero values, or else logz=True itself should fail on this plot, both when choosing the colors for the main plot and when making the colorbar itself.  Having it fail only when displaying the numerical values is odd.

**Comment by jlstevens** [comment]
> Having it fail only when displaying the numerical values is odd.

Good point. Whatever is being displayed with the colormap should have the corresponding value shown. I'm not sure how I would interpret log values when the data falls below 1 to be honest...

**Comment by jbednar** [comment]
I assume it's using [log1p](https://docs.scipy.org/doc/numpy-1.14.0/reference/generated/numpy.log1p.html), so the colorbar labelling can just do the same.
